server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -903,6 +903,7 @@ public static void remoteDirectoryCleanup(
     ) {
         try {
             RemoteSegmentStoreDirectory remoteSegmentStoreDirectory = (RemoteSegmentStoreDirectory) remoteDirectoryFactory.newDirectory(
+                remoteStoreRepoForIndex,
```

Comment:
here as well, as mentioned in above comment, we can create the same remoteDirectory instance which have access only to metadata repository.